src/fastcs/transports/epics/ca/ioc.py

```
@@ -335,3 +337,27 @@ def _add_alias(record: RecordWrapper, alias: str | None):
+async def _run_and_set_alarm(record, coro: Awaitable):
+    """Await `coro` and update `record`'s alarm state based on the outcome.
+
+    On success, clears the alarm (NO_ALARM). On any exception, raises the
+    record into MAJOR_ALARM. The exception itself is not re-raised or
+    logged here, since `AttrW.put` already logs it; this function's only
+    job is to reflect the outcome in the record's alarm status.
+    """
+    try:
+        await coro
+        _set_alarm(record, alarm.NO_ALARM)
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)